### ISSUE TITLE
addTabWithDragAndDrop throws "Cannot read property 'currentTarget' of undefined" error

### ISSUE THREAD

**Issue by charodziej** [caplin/FlexLayout#110]
I'm not sure what is the correct format to report issues here, so I will just explain what's wrong.

Before the [0.4.0](https://github.com/caplin/FlexLayout/commit/aecddaeec6acbb33b99b8e62a44ec7e8decf4660) update the `addTabWithDragAndDrop` was working perfectly fine. After updating to the latest version though, this function doesn't appear to have a chance of working correctly.
The error "Cannot read property 'currentTarget' of undefined" is thrown on the line [104 in file DragDrop.ts](https://github.com/caplin/FlexLayout/blob/aecddaeec6acbb33b99b8e62a44ec7e8decf4660/src/DragDrop.ts#L104). 
```javascript
const currentDocument = ((event!.currentTarget) as HTMLDivElement).ownerDocument;
```
The use of the non-null assertion operator in `event!.currentTarget` means TypeScript won't complain about it, but `event` is `undefined` in this case as stated in the `addTabWithDragAndDrop` declaration on line [472 in file Layout.tsx](https://github.com/caplin/FlexLayout/blob/aecddaeec6acbb33b99b8e62a44ec7e8decf4660/src/view/Layout.tsx#L472).
```javascript
  addTabWithDragAndDrop(dragText: string, json: any, onDrop?: () => void) {
    this.fnNewNodeDropped = onDrop;
    this.newTabJson = json;
    this.dragStart(
      undefined,   // this is the undefined event
      dragText,
      TabNode._fromJson(json, this.model!),
      true,
      undefined,
      undefined
    );
  }
```

I can't propose a simple fix or workaround to this issue, because there is no simple way to avoid the use of `.ownerDocument`. In the case of my project it could just default to the `document`, but it certainly wouldn't work if someone wanted to use the popout tabs.
Until this is fixed, I suggest using `addTabWithDragAndDropIndirect`, even though the user experience isn't as satisfying. 

I hope someone with more insight into this project finds a good solution, because this is the best layout library I've found so far!

**Comment by nealus** [comment]
Hi Adam, thanks to telling me about this issue, I've added a fix in version 0.4.3

**Comment by charodziej** [comment]
Thanks for the very quick fix! Just downloaded the update and it works perfectly in my project. Thanks for creating such a great library and maintaining it!
